lib/maxmind/geoip2/record/traits.rb

```diff
@@ -0,0 +1,184 @@
+
+require 'ipaddr'
+require 'maxmind/geoip2/record/abstract'
+
+module MaxMind::GeoIP2::Record
+  # Contains data for the traits record associated with an IP address.
+  #
+  # This record is returned by all location services and databases.
+  class Traits < Abstract
+    def initialize(record) # :nodoc:
+      super(record)
+      if !record.key?('network') && record.key?('ip_address') &&
+         record.key?('prefix_length')
+        ip = IPAddr.new(
+          format('%s/%d', record['ip_address'], record['prefix_length']),
+        )
+        # We could use ip.prefix instead of record['prefix_length'], but that
+        # method only becomes available in Ruby 2.5+.
+        record['network'] = format('%s/%d', ip.to_s, record['prefix_length'])
+      end
+    end
+
+    # The autonomous system number associated with the IP address. See
+    # Wikipedia[https://en.wikipedia.org/wiki/Autonomous_system_(Internet)].
+    # This attribute is only available from the City and Insights web service
+    # and the GeoIP2 Enterprise database. Integer but may be nil.
+    def autonomous_system_number
+      get('autonomous_system_number')
+    end
+
+    # The organization associated with the registered autonomous system number
+    # for the IP address. See
+    # Wikipedia[https://en.wikipedia.org/wiki/Autonomous_system_(Internet)].
+    # This attribute is only available from the City and Insights web service
+    # and the GeoIP2 Enterprise database. String but may be nil.
+    def autonomous_system_organization
+      get('autonomous_system_organization')
+    end
+
+    # The connection type may take the following  values: "Dialup",
+    # "Cable/DSL", "Corporate", "Cellular". Additional values may be added in
+    # the future. This attribute is only available in the GeoIP2 Enterprise
+    # database. String but may be nil.
+    def connection_type
+      get('connection_type')
+    end
+
+    # The second level domain associated with the IP address. This will be
+    # something like "example.com" or "example.co.uk", not "foo.example.com".
+    # This attribute is only available from the City and Insights web service
+    # and the GeoIP2 Enterprise database. String but may be nil.
+    def domain
+      get('domain')
+    end
+
+    # The IP address that the data in the model is for. If you performed a "me"
+    # lookup against the web service, this will be the externally routable IP
+    # address for the system the code is running on. If the system is behind a
+    # NAT, this may differ from the IP address locally assigned to it. This
+    # attribute is returned by all end points. String.
+    def ip_address
+      get('ip_address')
+    end
+
+    # This is true if the IP address belongs to any sort of anonymous network.
+    # This property is only available from GeoIP2 Precision Insights. Boolean.
+    def anonymous?
+      get('is_anonymous')
+    end
+
+    # *Deprecated.* Please see our GeoIP2 Anonymous IP database
```

Comment:
It is weird including a deprecated attribute in a new API, but I guess we are still providing this data.

Reply:
Hmm, good point. Let's drop it!